### ISSUE TITLE
nvidiaPackages.production: `libnvidia-pkcs11.so.$DRIVER_VERSION` missing

### ISSUE THREAD

**Issue by msanft** [NixOS/nixpkgs#392750]
### Nixpkgs version

- Unstable (25.05)

### Describe the bug

When running a container with pyTorch on version `2.6.0+cu124` with a H100 NVIDIA GPU on the currently-packaged `production` NVIDIA driver in nixpkgs, PyTorch tries to load the following library:

```
# excerpt from strace output
openat(AT_FDCWD, "$PATH/libnvidia-pkcs11.so.550.90.07", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

This file is missing as we delete it in the build of the production driver:

https://github.com/NixOS/nixpkgs/blob/a13cc0bc42a89a025f545c4a17f25e6300710f39/pkgs/os-specific/linux/nvidia-x11/builder.sh#L81-L82

While we should definitely not include OpenSSL 1.1 by default (e.g. for security reasons), we should still somehow address this incompatibility.

### Steps to reproduce

This can be reproduced with the following command:

```sh
strace -f -e openat python3 -c 'import torch; torch.cuda.is_available()'
```

### Expected behaviour

PyTorch should load correctly.

### Relevant log output

```console

```

### System metadata

```
# uname -a
Linux workload-textgen-0 6.11.0-nvidia-gpu-confidential #1-NixOS SMP Sun Sep 15 14:57:56 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

Not applicable otherwise, as the system does not include `nix` nor `nix-*` et al.

### Notify maintainers

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc
